File: jvm-runtime/testdata/java/passthrough/src/main/java/xyz/block/ftl/test/DidMapper.java
package xyz.block.ftl.test;

import java.nio.charset.StandardCharsets;

import ftl.gomodule.DidTypeAliasMapper;
import web5.sdk.dids.didcore.Did;

public class DidMapper implements DidTypeAliasMapper {

    @Override
    public Did decode(String bytes) {
        return Did.Parser.parse(bytes);
    }

    @Override
    public String encode(Did did) {
        return new String(did.marshalText(), StandardCharsets.UTF_8);
    }
}

File: jvm-runtime/testdata/kotlin/passthrough/src/main/kotlin/xyz/block/ftl/test/DidMapper.kt
package xyz.block.ftl.test

import ftl.gomodule.DidTypeAliasMapper
import web5.sdk.dids.didcore.Did


class DidMapper : DidTypeAliasMapper
{

  override fun decode(bytes: String): Did {
    return Did.Parser.parse(bytes)
  }

  override fun encode(did: Did): String {
    return String(did.marshalText())
  }

}
